code/modules/surgery/organs/organ_internal.dm

```diff
@@ -51,6 +51,9 @@ INITIALIZE_IMMEDIATE(/obj/item/organ)
 			foodtypes = RAW | MEAT | GROSS,\
 			volume = reagent_vol,\
 			after_eat = CALLBACK(src, .proc/OnEatFrom))
+
+	RegisterSignal(src, SIGNAL_ADDTRAIT(TRAIT_ON_ICE), /atom.proc/begin_freeze)
```

Comment:
this is a bad design pattern its confusing as to what the future programmer who has to work with your code is doing. youre essentially using adding this trait to an organ is a proc call which isnt clear at all unless someone finds these two lines.

Reply:
So, remove the signal and call the proc directly?